src/briefcase/platforms/linux/appimage.py

```diff
@@ -175,13 +175,21 @@ class LinuxAppImageCreateCommand(
     def output_format_template_context(self, app: AppConfig):
         context = super().output_format_template_context(app)
 
-        # Add the manylinux tag to the template context.
         try:
-            tag = getattr(app, "manylinux_image_tag", "latest")
             manylinux_arch = {
                 "x86_64": "x86_64",
                 "i386": "i686",
-            }[LinuxDeploy.arch(self.tools.host_arch)]
+                "aarch64": "aarch64",
+            }[LinuxDeploy.arch(self.tools)]
+        except KeyError:
+            manylinux_arch = "unknown"
+            self.logger.warning(
+                f"There is not a manylinux base image for {LinuxDeploy.arch(self.tools)}"
+            )
```

Comment:
Why set the arch as "unknown" and printing a warning, so the next step fails (implicitly) on the Docker image lookup, rather than straight up failing on the arch lookup?

Reply:
This one is a bit annoying....and I didn't want to introduce a bunch more changes to accommodate this better.

First, this code runs even if someone passes `--no-docker`. So, we can't error here if someone isn't even trying to use Docker. But....if this code doesn't run and they don't pass `--no-docker` next time, then the `Dockerfile` will definitely be hosed. (This may be best resolved by telling the user to run `briefcase create linux appimage` again since they want to use Docker now....but then we'd need to track they didn't use it the first time.)

Second, if someone isn't using the Briefcase template for AppImage, then the `Dockerfile` probably won't care about which specific manylinux image this logic determines anyway. So, I suppose we could assess the proper architecture for the name of the manylinux image after the code checks if `app.manylinux` exists.....but that just kinda moves the error condition around a bit and doesn't really change the underlying situation.

I don't think any of this is ideal....but I was mostly trying to avoid refactoring a bunch of this right now. Although, this really only presents because this is also introducing support for 32bit arm. If we only support aarch64, this complication (and others) goes away.

This is the balance I landed on given all this....open to other balancing points.

Reply:
Ah - I hadn't considered the `--no-docker` case. That makes sense; and I think it's a reasonable compromise under the circumstances. My only thought is whether it might be better to use the actual architecture, rather than "unknown" (i.e., still raise the warning as you are now, but generate `manylinux_2_28_armv7h` as the manylinux base). That way, you'll still get an explicit pre-warning, but the Dockerfile will be in a state that won't look "wrong" - and when docker fails, the error will be of the form "manylinux_2_28_armv7h does not exist", rather than "manylinux_2_28_unknown does not exist". Both are equally true, but the former is a little more explicit about what doesn't exist... and it *might* exist at some point in the future.... (I doubt it, but it's at least *possible*)